app/src/main/java/com/lagradost/cloudstream3/animeproviders/TocanimeProvider.kt

```
@@ -0,0 +1,177 @@
+                "Đã hoàn thành" -> ShowStatus.Completed
+                "Chưa hoàn thành" -> ShowStatus.Ongoing
+                else -> ShowStatus.Completed
+            }
+        }
+    }
+
+    override suspend fun getMainPage(): HomePageResponse {
+        val document = app.get(mainUrl).document
+
+        val homePageList = ArrayList<HomePageList>()
+
+        document.select("div#playlists > div").forEach { block ->
+            val header = block.selectFirst("h2")?.text()?.trim() ?: ""
+            val items = block.select("div.col-lg-3.col-md-4.col-6").map {
+                it.toSearchResult()
+            }
+            if (items.isNotEmpty()) homePageList.add(HomePageList(header, items))
+        }
+
+        return HomePageResponse(homePageList)
+    }
+
+    private fun Element.toSearchResult(): AnimeSearchResponse {
+        val title = this.selectFirst("h3 a")?.text()?.trim() ?: ""
+        val href = fixUrl(this.selectFirst("h3 a")!!.attr("href"))
+        val posterUrl = fixUrlNull(this.selectFirst("div.card-item-img")?.attr("data-original"))
+        val epNum = this.selectFirst("div.card-item-badget.rtl")?.text()?.let { num ->
+            if (num.contains("Preview")) {
+                num.filter { it.isDigit() }.toIntOrNull()?.minus(1)
+            } else {
+                num.filter { it.isDigit() }.toIntOrNull()
+            }
+        }
+        return newAnimeSearchResponse(title, href, TvType.Anime) {
+            this.posterUrl = posterUrl
+            addSub(epNum)
+        }
+
+    }
+
+    override suspend fun search(query: String): List<SearchResponse> {
+        val document = app.get("$mainUrl/content/search?t=kw&q=$query").document
+
+        return document.select("div.col-lg-3.col-md-4.col-6").map {
+            it.toSearchResult()
+        }
+
+    }
+
+    override suspend fun load(url: String): LoadResponse {
+        val document = app.get(url).document
+
+        val title = document.selectFirst("h1.title")?.text() ?: ""
```

🤨 this should throw an error or return null